File: codeforces/1150/A.cpp
#include <iostream>
using namespace std;

int main() {
  int n, m , r;
  int val;
  int min_s = 1000, max_b = 0;
  cin >> n >> m >> r;
  for (int i = 0; i < n; i++) {
    cin >> val;
    if (min_s >  val) {
      min_s = val;
    }
  } 

  for (int i = 0; i < m; i++) {
    cin >> val;
    if (max_b <  val) {
      max_b = val;
    }
  }
  int maxValue = (r / min_s) * max_b + (r % min_s);
  if (maxValue > r) {
    cout << maxValue;
  }
  else {
    cout << r;
  }
  return 0;
}

File: codeforces/1150/A.py
n, m, r = list(map(int, input().strip().split(" ")))
s = list(map(int, input().strip().split(" ")))
b = list(map(int, input().strip().split(" ")))
min_s = min(*s)
max_b = max(*b)

maxValue = max((r // min_s) * max_b + (r % min_s), r)
print(maxValue)
